src/js/components/sources.jsx

```diff
@@ -0,0 +1,148 @@
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import * as ArticleAction from '../actions/ArticleAction';
+import sourceStore from '../stores/SourceStore';
+import Header from './Header.jsx';
+
+
+export default class Sources extends Component {
+
+  constructor() {
+    super();
+
+    this.state = {
+      sources: [],
+      searchInput: ''
+    };
+    /**
+     * We need to bind this to onChange so we can have
+     * the proper this reference inside the method
+     */
+    this.onChange = this.onChange.bind(this);
```

Comment:
no need to bind here. refer to comment made about this in `src/js/components/Header.jsx` file